py/modmicropython.c

```diff
@@ -130,9 +130,20 @@ STATIC MP_DEFINE_CONST_FUN_OBJ_0(mp_micropython_heap_lock_obj, mp_micropython_he
 
 STATIC mp_obj_t mp_micropython_heap_unlock(void) {
     gc_unlock();
-    return mp_const_none;
+    return MP_OBJ_NEW_SMALL_INT(MP_STATE_MEM(gc_lock_depth));
 }
 STATIC MP_DEFINE_CONST_FUN_OBJ_0(mp_micropython_heap_unlock_obj, mp_micropython_heap_unlock);
+
+#if MICROPY_PY_ENABLE_HEAP_LOCKED
+STATIC mp_obj_t mp_micropython_heap_locked(void) {
+    // check if GC is locked
+    if (MP_STATE_MEM(gc_lock_depth) > 0) {
+        return mp_const_true;
+    }
+    return mp_const_false;
```

Comment:
Might be able to reduce size of the function by using:
```c
return MP_OBJ_NEW_SMALL_INT(MP_STATE_MEM(gc_lock_depth));
```
That would still work the same (returns 0 aka false if not locked) and have the added bonus that you can also determine the lock depth.

Otherwise, `mp_obj_new_bool(MP_STATE_MEM(gc_lock_depth) > 0)` is a more concise way of writing this.